---
title: "Multiparity_Comparison"
author: "Kevin Jou"
date: "2023-08-10"
output: html_document
---
# Introduction
Multiparity (giving birth >=3 times) is known to have an associated risk factor with diabetes, however the exact cause of this correlation is not well characterized.  In an attempt to better understand the potential changes occurring in the insulin producing islets, a lab in South Korea has taken isolated islets from a 9 week old virgin mouse and a multiparous mouse that had undergone 3 consecutive pregnancies.[1]  These islets were then transplanted via kidney capsule into a recipient mouse which then underwent S-961 injection to stimulate diabetic conditions.  The islets were removed from the mouse after a currently unknown time period and then analysed using scRNAseq on the HiSeq X Ten.

Reference: [1] Lee et. al., 2023, Multiparity increases the risk of diabetes by impairing proliferative capacity of beta cells II(data accessible ad NCBI GEO database(Edgar et al.,2002), accession GSE234741))

# Materials
R version 4.3.0
RStudio
Seurat library
dplyr libary
RaceID library

# Methods
To analyse the islets and compare their relative expression and visualize it, we will largely be using the Seurat package as it can read our matrix and then create a Seurat data object for us to attach annotations and other analyses to.

Both datasets for virgin and multiparous mice will then be processed in parallel for UMAP projection and stemID analysis

Both datasets will then undergo QC to remove any low quality count data from the dataset which may influence the results.
After QC, the neighborhood association for the datasets will be calculated based on the ideal number of Principle Components before being graphed with UMAP.

Clutsers defined by the louvain algorithm will then be compared based on marker gene expression and grouped together to see if there is any noticeable difference in expression across datasets.

Matrix will then be processed using RaceID library to predict which clusters may contain stem cells.  Datasets will be compared based on average score calculated using links * delta entropy.

## Load necessary libraries
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(Seurat)
library(dplyr)
library(RaceID)
```

## Load Data and Create Seurat Objects for each condition
```{r}
virgin.data <- Read10X(data.dir="C:/Users/Kevin/Documents/Summer semester/virgin")
mp.data <- Read10X(data.dir="C:/Users/Kevin/Documents/Summer semester/multiparous")
virginislet <- CreateSeuratObject(counts = virgin.data, project = "virgin")
MPislet <- CreateSeuratObject(counts= mp.data, project = "multiparous")
```
Data is loaded into matrix using Read10X function since output contains 3 files with barcodes, features, and genecount.  Seurat object is then created using matrix data as count data.
## Mitochondrial Annotation
```{r}
virginislet[["percent.mt"]] <- PercentageFeatureSet(virginislet, pattern = "^mt-")
MPislet[["percent.mt"]] <- PercentageFeatureSet(MPislet, pattern = "^mt-")
```
Mitochondrial genes are annotated for QC purposes.
## Visualization
```{r}
VlnPlot(virginislet, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), split.by="orig.ident", pt.size=0.1)
VlnPlot(MPislet, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), split.by="orig.ident", pt.size=0.1)
```
### Visualization of QC metrics for filtering data.  High percent mitochondrial samples are considered low quality and should be removed pre-processing.
## Filter Data
```{r}
min_gene <- 200
max_gene <- 5000
mt_percent <- 15
virginislet <- subset(virginislet, subset = nFeature_RNA > min_gene & percent.mt < mt_percent & nFeature_RNA < max_gene)
MPislet <- subset(MPislet, subset = nFeature_RNA > min_gene & percent.mt < mt_percent & nFeature_RNA < max_gene)
```
Filter out cells based on data you want to capture.  In this case we only want cells with more than 200 gene counts but less than 5000, and less than 15% of the total gene contributions must be from mitochondrial RNA.

```{r}
VlnPlot(virginislet, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), split.by="orig.ident", pt.size=0.1)
VlnPlot(MPislet, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), split.by="orig.ident", pt.size=0.1)
```
### Visualiztion of QC metrics after filtering cells to ensure filtering was successfully performed

## Run QC
```{r}
virginislet <- ScaleData(virginislet, verbose = FALSE)
virginislet <- NormalizeData(virginislet)
virginislet <- FindVariableFeatures(virginislet, selection.method = "vst", nfeatures = 2000, verbose = FALSE)

MPislet <- ScaleData(MPislet, verbose = FALSE)
MPislet <- NormalizeData(MPislet)
MPislet <- FindVariableFeatures(MPislet, selection.method = "vst", nfeatures = 2000, verbose = FALSE)
```
To make the data easier to visualize and compare, the data will be scaled and normalized so that the means of all of the genes are set at 0 and the number of counts across all genes is equal.  This allows for easier relative comparison across genes as well as graphing them in a reasonably scaled graph.
## PCA
```{r}
virginislet <- RunPCA(virginislet, verbose = FALSE)

MPislet <- RunPCA(MPislet, verbose = FALSE)

ElbowPlot(virginislet,ndim=30)
ElbowPlot(MPislet,ndim=30)
```
In order to identify the ideal number of Principle Components to consider for dimensionality reduction, we use Seurat's elbowplot to see how much of the variation is captured by each Principle Comonent.  A lower STDev value is better.
The elbow plot shows that 20 PCs is likely enough to capture a majority of the variation in both groups.
## UMAP
```{r}
virginislet <- FindNeighbors(virginislet, dims=1:20)
MPislet <- FindNeighbors(MPislet, dims=1:20)

virginislet <- FindClusters(virginislet)
MPislet <- FindClusters(MPislet)

virginislet <- RunUMAP(virginislet, dims = 1:20)
MPislet <- RunUMAP(MPislet, dims = 1:20)
```
Neighborhood graph created to calculated distance from nearest neighbor in dataset before undergoing clustering via the Louvain algorithm.  Once the clusters are identified, the clusters will be displayed in 2 dimensions using UMAP to reduce the dimensionality of the data while still accounting for a majority of the variability in the datasets. 
```{r}
DimPlot(virginislet)
DimPlot(MPislet)
```
### Visualization of the clusters and their relative positions to one another based on gene expression.
## Marker Analysis
```{r}
virginmarkers <- FindAllMarkers(virginislet, only.pos=TRUE)
MPmarkers <- FindAllMarkers(MPislet, only.pos=TRUE)
```
Identify upregulated marker genes in each cluster.
```{r}
virginmarkers %>% group_by(cluster) %>% top_n(15, avg_log2FC)
MPmarkers %>% group_by(cluster) %>% top_n(15, avg_log2FC)
```
Sort dataframe to output top 15 upregulated genes by log2fc value.  Can view to potentially identify marker genes.  Since this dataset is already using an isolated cell type, we expect most of these genes to correlate to endocrine cell types.

```{r}
VlnPlot(virginislet,features=c("Ins2","Gcg","Sst","Ppy","Ghrl","Krt19","Rgs5","Sds","Tpsab1","Sox10"),pt.size=0.25)
DotPlot(virginislet,features=c("Ins2","Gcg","Sst","Ppy","Cpa1"))
DotPlot(virginislet,features=c("Krt19","Nkx6-1","Pdx1","Sox9", "Cd52", "Klk1", "Muc1"))

VlnPlot(MPislet,features=c("Ins2","Gcg","Sst","Ppy","Ghrl","Krt19","Rgs5","Sds","Tpsab1","Sox10"),pt.size=0.25)
DotPlot(MPislet,features=c("Ins2","Gcg","Sst","Ppy","Cpa1"))
DotPlot(MPislet,features=c("Krt19","Nkx6-1","Pdx1","Sox9", "Cd52", "Klk1", "Muc1"))
```
### DotPlot and Violin plot of marker genes of interest to be compared for identifying cell types present.
```{r}
vcurrent.cluster.ids <- names(table(virginislet@active.ident))
vnew.cluster.ids <- c("endocrine2","endocrine1","endocrine1","endocrine2","endocrine2","endocrine3","endocrine3","endocrine3","endocrine3","immune","immune","endocrine","immune","endocrine","gamma","endocrine4", "endocrine4", "endocrine4", "endocrine4")
virginislet@active.ident <- plyr::mapvalues(x = virginislet@active.ident, from = vcurrent.cluster.ids, to = vnew.cluster.ids)

mpcurrent.cluster.ids <- names(table(MPislet@active.ident))
mpnew.cluster.ids <- c("endocrine1","endocrine1","endocrine1","endocrine2","endocrine3","endocrine4","endocrine1","endocrine4","endocrine4","endocrine4","immune","gamma","endocrine5","immune","immune","endocrine5", "immune", "endocrine5", "endocrine5")
MPislet@active.ident <- plyr::mapvalues(x = MPislet@active.ident, from = mpcurrent.cluster.ids, to = mpnew.cluster.ids)
```
Clusters are re-labeled based on identification in previous step.
```{r}
DimPlot(virginislet)
DimPlot(MPislet)
```
### UMAP of clusters after identification

```{r}
VlnPlot(virginislet,features=c("Ins2","Gcg","Sst","Ppy","Ghrl","Krt19","Rgs5","Sds","Tpsab1","Sox10"),pt.size=0.25)
DotPlot(virginislet,features=c("Ins2","Gcg","Sst","Ppy","Cpa1"))
DotPlot(virginislet,features=c("Cdk2","Nkx6-1","Pdx1","Sox9", "Cd52", "Itga6", "Prom1"))

VlnPlot(MPislet,features=c("Ins2","Gcg","Sst","Ppy","Ghrl","Krt19","Rgs5","Sds","Tpsab1","Sox10"),pt.size=0.25)
DotPlot(MPislet,features=c("Ins2","Gcg","Sst","Ppy","Cpa1"))
DotPlot(MPislet,features=c("Krt19","Nkx6-1","Pdx1","Sox9", "Cd52", "Itga6", "Prom1"))
```
### DotPlot and Violin plot of marker genes and other genes of interest for comparison after identification.

## StemID
StemID analyses takes output from RaceID2.  It is thus necessary to use the original count matrix to process the data as that is what is the expected raw data for input into RaceID.
```{r}
sc <- SCseq(virgin.data)
sc <- filterdata(sc, mintotal = 2000)
sc <- compdist(sc,metric="pearson")
sc <- clustexp(sc)
plotsaturation(sc,disp=TRUE)
plotjaccard(sc)
```
Create SCseq object using matrix data loaded in from Seurat package.  Filter data with less than 2000 counts across the entire transcript.  Cells are then clustered using RaceIDs built in algorithm using default clusters k=30 as a test case.

Generated graphs show default cluster number is an overestimate, leading to instability as indicated by low Jaccard’s similarity. Based on visual inspection of the average within-cluster dispersion as a function of the cluster number, we manually set the cluster number to 16 without recomputing the saturation behavior.
```{r}
sc <- clustexp(sc,cln=16,sat=FALSE)
```

```{r}
sc <- findoutliers(sc)
plotoutlierprobs(sc)
```
For visualization of outliers in the data and how they may be affecting the dataset.
```{r}
sc <- compumap(sc)
sc <- compfr(sc)
sc <- comptsne(sc)
```
Dimensional reduction methods are perfomed on dataset for use in StemID tree analyses.

```{r}
ltr <- Ltree(sc)
ltr <- compentropy(ltr)
ltr <- projcells(ltr,cthr=5,nmode=FALSE, fr=TRUE)
ltr <- projback(ltr,pdishuf=30)
```
StemID links are generated using Fruchterman-Rheingold representation and then computation is performed 30(>20 ideally) times to ensure quality of data.

```{r}
ltr <- lineagegraph(ltr)
ltr <- comppvalue(ltr)
```

```{r}
plotgraph(ltr,scthr=0.2,showCells=FALSE,showMap=TRUE)
```
## Plot showing potential source of stem cells among clusters is shown here.

```{r}
x <- compscore(ltr,scthr=0.2)
plotspantree(ltr)
```
Scores of each cluster are calculated here for StemID.  Higher score indicates more likeliness to be a stem cell in the population.
## Multiparous StemID
Repeat steps performed on virgin dataset for multiparous dataset.
```{r}
mp <- SCseq(mp.data)
mp <- filterdata(mp, mintotal = 2000)
mp <- compdist(mp,metric="pearson")
mp <- clustexp(mp)
plotsaturation(mp,disp=TRUE)
plotjaccard(mp)
```

```{r}
mp <- clustexp(mp,cln=16,sat=FALSE)
```

```{r}
mp <- findoutliers(mp)
plotoutlierprobs(mp)
```

```{r}
mp <- compumap(mp)
mp <- compfr(mp)
mp <- comptsne(mp)
```

```{r}
ltm <- Ltree(mp)
ltm <- compentropy(ltm)
ltm <- projcells(ltm,cthr=5,nmode=FALSE, fr=TRUE)
ltm <- projback(ltm,pdishuf=30)
```

```{r}
ltm <- lineagegraph(ltm)
ltm <- comppvalue(ltm)
```

```{r}
y <- compscore(ltm,scthr=0.2)
plotspantree(ltm)

x <- compscore(ltr,scthr=0.2)
plotspantree(ltr)
```

# Analysis
Based on the analyses of the datasets, there appear to be some differences in expression among the virgin and multiparous endocrine cells. One notable gene SST has negative expression in the gamma cells of the virgin sourced islets but a positive expression in the multiparous islets.  There are some other small differences such as a higher expression of double positive Prom1 Itga6 cells which are thought to correlate to progenitor like cells in the pancreas.[2]

The StemID analysis comparison between the two datasets is more interesting as you can see there are higher scoring clusters in the virgin islets compared to the multiparous islets  This may imply a greater proportion of stem cells exist in the virgin islets than the multiparous islets, which may somewhat explain the increased risk of diabetes if the islets are unable to effectively regenerate in multiparous individuals.  Further study with longer time points may show a more complete picture depending on if the mice take longer than 9 weeks to develop a distinct phenotype.

Overall the results do agree with the initial summary performed by the lab in Korea, but there is still no strong gene marker or group of markers yet identified which may explain some of the changes in function after multiple births.

Reference: [2]Tremblay et al., “Identification of a Distinct Ductal Subpopulation with Self-Renewal and Differentiation Potential from the Adult Murine Pancreas.”

#Conclusion
There are notable differences in relative expression among the multiparous and virgin islets, including marker genes such as Prom1 and Itga6 which are theorized to correlated to progenitor like cells.  The stemID analysis of the virgin islet clusters and multiparous islet clusters shows a clear increase in stem cell score in the virgin islets compared to the multiparous islets.  This information collected from relative expression and algorithm prediction may point to the reduced function of stem cells / progenitor cells in correlation to reduced islet function, although further studies would need to be performed in order to strengthen the statistical basis of this assumption.  Overall, the trends match with what the original source of data from GSE was claiming about the islets.
