---
title: "Computational Biology Final"
author: "Kevin Jou"
date: "2024-05-05"
output: html_document
---

```{r}
knitr::opts_chunk$set(echo = TRUE)
library(maftools)
```

```{r}
### Load Pancreatic adenocarcinoma donor datasets
ds1 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/d7e3f62c-1c5b-4324-b869-7c84d04474d4/e34ef06f-5333-46a5-8bf8-30d1354767c9.wxs.aliquot_ensemble_masked.maf")
ds2 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/79f27ad8-582c-4716-9af6-dabfc3fe49f5/f2cec1b1-11da-43e0-be7a-86f8651740b9.wxs.aliquot_ensemble_masked.maf")
ds3 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/1b50edae-a3e6-47f1-96e3-f65bdc37f76a/e422924e-7fba-4fda-b1a0-633b1925d114.wxs.aliquot_ensemble_masked.maf")
ds4 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/70f60875-638c-4d6c-b423-ca14f465475e/d9d6133c-35d1-4c5d-853e-26881f327ea6.wxs.aliquot_ensemble_masked.maf")
ds5 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/399fbaeb-9ddd-4b89-890b-2542fe7dd89f/835eefb4-a353-4073-a35d-01326f2684a8.wxs.aliquot_ensemble_masked.maf")
ds6 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/192621c3-b43f-47da-ad60-da63afb0d8b8/b762baa6-f35d-4bde-a9fa-5adcb204f5b3.wxs.aliquot_ensemble_masked.maf")
ds7 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/35286188-bd30-4f82-a5fb-cbb1020bdf24/6209b5f4-30b5-4add-891b-b56e9c0272f3.wxs.aliquot_ensemble_masked.maf")
ds8 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/b5842b7b-8c9f-4a4c-bd4e-e19b30fbc714/01eca97d-aa1b-4059-b70b-6aadfb82012e.wxs.aliquot_ensemble_masked.maf")
ds9 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/c2ed2536-d87a-44ff-ab2c-f15c57197a4b/a3b2094f-c2f0-48a8-b7e6-0612409a962e.wxs.aliquot_ensemble_masked.maf")
ds10 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/c5b27701-40c9-447c-9dd1-5f53c8cb5817/467dbc81-92e4-477c-880f-96bad777e21a.wxs.aliquot_ensemble_masked.maf")
ds11 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/d0586a8c-388c-410e-96c2-de192a62cc66/b714d8da-5709-4104-8eb6-03ba111066db.targeted_sequencing.aliquot_ensemble_masked.maf")
ds12 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/e4535f8c-63e9-4d68-821f-2687a8704259/b4c7b22f-fd99-4246-b771-9a9486855e07.wxs.aliquot_ensemble_masked.maf")
ds13 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/f18aae45-aad4-4e3c-85c1-0e8671cdd13b/14bdda45-1f89-4ffb-a1c3-b79a5d99c35b.wxs.aliquot_ensemble_masked.maf")
ds14 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/2c51a6a4-2873-4716-ab23-248386ce2cec/462ff53d-8a58-4bc8-8a4b-c36f79f2c361.wxs.aliquot_ensemble_masked.maf")
ds15 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/7b9d1e93-f9b6-4e17-963c-a098a573d8d7/884dd8cd-185f-4756-aac2-4adc4e673149.wxs.aliquot_ensemble_masked.maf")
ds16 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/8c2214bf-daa9-4ea5-bf48-8c0cb0eba599/a6a54a9c-b844-4a45-ae01-451a61bafc19.wxs.aliquot_ensemble_masked.maf")
ds17 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/78fffa10-5888-4985-95c4-64fd2727fcda/0aa4babc-0373-4771-a07f-9083685d04db.wxs.aliquot_ensemble_masked.maf")
ds18 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/085b24cb-412c-411c-bc20-fb211db8703b/779f7624-6c69-46e8-b4e9-2bb27de6216b.wxs.aliquot_ensemble_masked.maf")
ds19 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/4853b47f-d1cb-4dd4-80d5-d230c81e7ca0/ed2b3215-a2d9-419f-932b-0071b103f83d.wxs.aliquot_ensemble_masked.maf")
ds20 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/a38da735-8086-409f-9a4e-ad3fa39d3a16/ed7aa83c-47a8-4c75-9b42-69826203a93b.wxs.aliquot_ensemble_masked.maf")
ds21 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/b7058028-40b4-4ee5-9eff-6a6df5f47904/37d4fc49-5069-4ed6-938f-9016a11658af.wxs.aliquot_ensemble_masked.maf")
ds22 <- read.maf("C:/Users/Kevin/Documents/Spring Semester/c70f389b-b2d6-49d7-8702-dff2dc8c55e8/41869091-1d06-49fe-8dcf-2f7008202f78.wxs.aliquot_ensemble_masked.maf")
```
```{r}
#Consolidate each loaded maf into one file for handling with the TRONCO library
mafs <- c(ds1,ds2,ds3,ds4,ds5,ds6,ds7,ds8,ds9,ds10,ds11,ds12,ds14,ds15,ds16,ds17,ds18,ds18,ds19,ds20,ds21,ds22)
save(mafs, file = "PDACdonormafs.RData")
mergemaf <- merge_mafs(mafs)
```
```{r}
#Write new pancreasmerge.maf file for use in TRONCO library
write.mafSummary(maf=mergemaf, basename = "pancreasmerge")
```

```{r}
library(TRONCO)
```


```{r echo=T, results='hide'}

#Import created MAF file in TRONCO
Mergedmaf <- import.MAF("C:/Users/Kevin/Documents/Spring Semester/pancreasmerge_maftools.maf", merge.mutation.types = FALSE, silent=TRUE)
as.genotypes(Mergedmaf)
```
```{r}
#Load head to check if data loaded properly
as.events(Mergedmaf)[1:20, ]
```
```{r}
oncoprint(Mergedmaf)
```

```{r}
#Filter gene events by frequency in number of samples,  due to low N I used a relatively strict gating of 20% occurrence, or about 4 samples
alterations = events.selection(as.alterations(Mergedmaf), filter.freq = .2)
```
```{r}
#Annotate genes involved in canonical pathway
gene.hypotheses = c('SMAD4', 'KRAS', 'CDKN2A', 'TP53')
Mergedmaf.clean = events.selection(Mergedmaf,
    filter.in.names=c(as.genes(alterations), gene.hypotheses))
```
```{r}
#Check data for duplicates
duplicates(Mergedmaf.clean)
```

```{r}
oncoprint(Mergedmaf.clean, gene.annot = list(priors = gene.hypotheses), sample.id = TRUE, font.column = 10, font.row = 5)
```

```{r}
#Check data for any data with input that is not readable by CAPRI or CAPRESE
consolidate.data(Mergedmaf.clean)
```
```{r}
#Clean indistinguishable events from the maf file
Mergedmaf.clean2 <- Mergedmaf.clean
```

```{r}
silent_genes <- c("MUC16", "CELSR2", "KMT2C", "TG", "PDZRN3", "DOCK2")
for (m in silent_genes)
  {
  Mergedmaf.clean2 <- delete.event(Mergedmaf.clean2, gene=m, type="Silent")
}
nonsgenes <- c("DOCK2", "MUC16")
for (n in nonsgenes)
{
  Mergedmaf.clean2 <- delete.event(Mergedmaf.clean2, gene=n, type="Nonsense_Mutation")
}

fs_genes <- c("KMT2D", "CDKN2A", "SMAD4")
for (v in fs_genes)
{
  Mergedmaf.clean2 <- delete.event(Mergedmaf.clean2, gene=v, type="Frame_Shift_Del")
}

Mergemaf.clean2 <- delete.event(Mergedmaf.clean2, gene="CDKN2A", type="In_Frame_Del")
```

```{r}
consolidate.data(Mergedmaf.clean2)
```


```{r}
#Create model using algorithms provided in TRONCO library
model <- tronco.capri(
  Mergedmaf.clean2,
  boot.seed = 300,
  nboot = 5
)
```
```{r}
model = annotate.description(model, 'CAPRI - PDAC')
```

```{r}
tronco.plot(model, 
    fontsize = 10, 
    confidence = c('tp', 'pr', 'hg'), 
    height.logic = 1, 
    legend.cex = 0.5, 
    pathways = list(priors = gene.hypotheses), 
    label.edge.size = 5)
```

```{r}
model.caprese = tronco.caprese(Mergedmaf.clean2)
model.caprese = annotate.description(model.caprese, 'CAPRESE - PDAC')
oncoprint(model)
```

```{r}
tronco.plot(model.caprese, 
    fontsize = 10, 
    confidence = c('tp', 'pr', 'hg'), 
    height.logic = 1, 
    legend.cex = 0.5, 
    pathways = list(priors = gene.hypotheses), 
    label.edge.size = 5)
```
```{r}
model.edmonds = tronco.edmonds(Mergedmaf.clean2)
model.edmonds = annotate.description(model.edmonds, 'EDMONDS - PDAC')

model.chowliu = tronco.chowliu(Mergedmaf.clean2)
model.chowliu = annotate.description(model.chowliu, 'CHOWLIU - PDAC')

model.prim = tronco.prim(Mergedmaf.clean2)
model.prim= annotate.description(model.prim, 'PRIM - PDAC')
```

```{r}
tronco.plot(model.edmonds, 
    fontsize = 10, 
    confidence = c('tp', 'pr', 'hg'), 
    height.logic = 1,
    scale.nodes= 1,
    legend.cex = 0.5, 
    pathways = list(priors = gene.hypotheses), 
    label.edge.size = 5)
```

```{r}
tronco.plot(model.chowliu, 
    fontsize = 10, 
    confidence = c('tp', 'pr', 'hg'), 
    height.logic = 1, 
    legend.cex = 0.5, 
    pathways = list(priors = gene.hypotheses), 
    label.edge.size = 5)
```

```{r}
tronco.plot(model.prim, 
    fontsize = 10, 
    confidence = c('tp', 'pr', 'hg'), 
    height.logic = 1, 
    legend.cex = 0.5, 
    pathways = list(priors = gene.hypotheses), 
    label.edge.size = 5)
```

